backend/lib/fs_file_storage.py

```diff
@@ -14,7 +16,11 @@ def save_file_obj(dest_path, file_obj, container, content_type):
         copyfileobj(file_obj, dest_file)
 
     uri = '{}{}/{}'.format(settings.MEDIA_URL, container, dest_path)
-    return settings.INTERNAL_MEDIA_HOST + uri, site.build_full_url(uri)
+    internal_url = new_signed_url(settings.INTERNAL_MEDIA_HOST + uri)
```

Comment:
Was the design to sign the URL when the file is saved, or when the file is to be fetched? I am under the impression that we agreed signing the URL when the file is to be fetched is a more elegant solution. If so, shouldn't this method return the unsigned URL?

Reply:
The [SignedHmacUrlSerializer](https://github.com/TheSpaghettiDetective/obico-server/pull/828/files#diff-03996b8fbe7d2c1ef16ec45987f317b42e0baca33f90b4d988f89cd173684bb8R31-R65) class signs (or re-signs) URLs as they are retrieved from the API.

However, there are several places in the code that save a file object and then pass the returned internal or external URL through to ml_api, tasks, redis, or the web frontend, where it is immediately fetched.
![image](https://github.com/TheSpaghettiDetective/obico-server/assets/91706106/306db9ee-1e8a-45de-9a80-b9086a2af4b8)

Since these URLs don't go through the API/serializer, we need to make sure they are signed before they get passed to other locations. 

I picked this spot because it:
1. ensures the URL will always be signed, even if it doesn't go through the API/serializer/database
2. avoids the need to call `new_signed_url()` in all downstream functions (and any future code changes)
3. if I understand correctly, `fs_file_storage.py` is only loaded for self-hosted instances, so it won't affect the cloud instance.

However, if you think there is an advantage to signing the URLs in the downstream locations then I'm happy to make the changes. Let me know what you think.

Reply:
The original design was:

`internal_url`: the url that can only be accessed by other containers on the same docker network. It shouldn't go through the `build_full_url` method. Instead, it should use `settings.INTERNAL_MEDIA_HOST`, which is `http://web:3334` by default, which is only accessible from other containers, such as "tasks". 

`external_url`: the url that will needs to be accessible from the internet, such as browser or email.

Can you make sure all the downstream code that directly uses `internal_url` all happens within one of the containers, namely "web", "tasks", "ml_api"? If so, then `internal_url` doesn't need to be signed. And I'd like to keep this design consistent with before (and should be simpler too).